omnigibson/systems/system_base.py

```diff
@@ -208,6 +208,9 @@ def _clear(self):
         if og.sim.is_playing() and gm.ENABLE_TRANSITION_RULES:
             self.scene.transition_rule_api.prune_active_rules()
 
+        # Clear from scene active system registry
+        self.scene.system_registry.remove(self)
+
```

Comment:
we should probably move all these logic into `remove_system` of scene?

Reply:
done